### PR TITLE
Dev branch sync

### PR DESCRIPTION
**Pull request by cone-forest**

### CODE REVIEW

File: src/renderer/resources/descriptor/descriptor.cpp [hunk 1]
```diff
@@ -0,0 +1,116 @@
+#include "resources/descriptor/descriptor.hpp"
+
+void mr::Descriptor::apply() {}
+
+mr::Descriptor::Descriptor(const VulkanState &state, Pipeline *pipeline, const std::vector<DescriptorAttachment> &attachments, uint set_number)
+  : _set_number(set_number), _set_layout(pipeline->set_layout(_set_number))
```
Comment (mitya-y) [inline src/renderer/resources/descriptor/descriptor.cpp:6]: This is crazy

**Review by mitya-y — CHANGES_REQUESTED** [review]
Corresponding issues created

**Review by dantibel — CHANGES_REQUESTED** [review]
Close issues

**Review by cone-forest — COMMENTED** [review]
Issues submitted

**Review by mitya-y — APPROVED** [review]
'main' branch will contain the most modern working version (which compiles and draws the scene after starting) of the engine. Currently, the scene is simple, demonstrating two planes with textures and a depth buffer.

**Pull request merged**